libraries/AP_HAL_Linux/SPIDevice.cpp

```diff
@@ -426,6 +426,15 @@ SPIDeviceManager::get_device(const char *name)
     return dev;
 }
 
+uint8_t SPIDeviceManager::get_count() {
+   return _n_device_desc;
+}
+
+void SPIDeviceManager::get_device_name(const int ref, char *name)
+{
+    strcpy(name, _device[ref].name);
```

Comment:
`strcpy` should be in the hall of "don't use functions" unless you really know what you are doing (and in this cases, you should probably use memcpy).

The reason is that it will copy any number of bytes (the string length) to the param name. The caller has no idea what's the size of the buffer he has to declare in order to accommodate the string. And the function doesn't check if there's enough space.

So, if this function does proves itself worthy, it should be changed to: 
```
const char *SPIDeviceManager::get_device_name(uint8_t n)
```

Then you don't copy anything, you just return a const pointer (so the user doesn't modify it, but copy himself if he needs to).